### ISSUE TITLE
Improve User XP on multiple services

### ISSUE THREAD

**Issue by mloiseleur** [traefik/traefik#10258]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you expect to see?

# Current user XP

On docker provider, a _guessing_ system link _router_ with _service_ when there is only one service.
It has been introduced 4 years ago with [this PR](https://github.com/traefik/traefik/pull/4621).

It means that it works with:

```yaml
labels:
- "traefik.http.routers.service1.rule=host(service1.docker.localhost)"
- "traefik.http.services.service1.loadbalancer.server.port=8381"
```

When adding another service, it fails with `too many services error` until user figures out he needs to explicitly link router with the service:

```yaml
- "traefik.http.routers.router1.service=service1"
```

Many users encountered it:

- https://github.com/traefik/traefik/issues/4807
- https://community.traefik.io/t/https-h2c-could-not-define-the-service-name-for-the-router-too-many-services/7092/2
- https://community.traefik.io/t/could-not-define-the-service-name-for-the-router-too-many-services/5514
- https://community.traefik.io/t/too-many-services-with-more-subdomains/15865
- https://community.traefik.io/t/service-router-wont-get-created/6769
- https://community.traefik.io/t/https-middleware-not-firing-for-certain-stack/3284
- https://community.traefik.io/t/bad-gateway-on-portainer/16384/3
- https://community.traefik.io/t/traefik-with-lots-of-routes-in-one-docker-service/9270
- https://community.traefik.io/t/2-services-in-one-docker-one-hides-the-other/6651

This issue is opened to discuss how to improve the current situation.

# Considered solutions

A. **Remove the guessing system**

It's a breaking change. There will be less magic (one more label / config line needed) and also a more consistent user XP.

B. **Rework guessing system using a convention**

Following [Convention over Configuration](https://en.wikipedia.org/wiki/Convention_over_configuration) paradigm, the guessing system could use a match in the name between the router and the service to make the link.

**Comment by bluepuma77** [comment]
Why a breaking change? This will break a lot of installations as it is used a lot from what I read on the community forum.

How about just making the error message more extensive and explaining the situation?

**Comment by mloiseleur** [comment]
I agree with you: improving the error message seems to be a simple good step on this.

I observe also that current example in the doc do not explicitly link service with the router. For instance [here](https://doc.traefik.io/traefik/routing/providers/docker/#tcp) or [here](https://doc.traefik.io/traefik/routing/providers/docker/#udp).

Wdyt about also change examples in the doc with an explicit link ?

**Comment by emilevauge** [comment]
Hi,
After discussing with the maintainer team, and following my comment on the [this issue](https://github.com/traefik/traefik/pull/10252#issuecomment-2037542029), here is our statement on this topic: we do agree that the "too many services" error message is too vague, and that the documentation should be improved.
However, we don't think changing the current implmentation makes sense, the fact that it brakes the compatibility outweigh the benefits brought by this change.

Our current plan is to:

- clarify the error message
- improve the corresponding documentation
